src/gpu/intel/jit/dsl/dsl.cpp

```diff
@@ -71,19 +92,25 @@ struct ctx_t {
 
             for (int i = 0; i < 3; i++) {
                 group_ids_[i]
-                        = let(group_id_type(), ir_builder_t::tg_idx(i), {});
+                        = let(ir_builder_t::tg_idx(i), group_id_type(), {});
                 local_ids_[i]
-                        = let(local_id_type(), ir_builder_t::local_id(i), {});
+                        = let(ir_builder_t::local_id(i), local_id_type(), {});
                 local_sizes_[i] = let(
-                        local_size_type(), ir_builder_t::local_size(i), {});
+                        ir_builder_t::local_size(i), local_size_type(), {});
             }
         }
     }
 
     kernel_t end_kernel() {
         gpu_assert(stmts_stack_.size() == 1)
                 << "Invalid end of kernel, imbalanced scopes detected";
-        kernel_t ret {std::move(interface_), pop_scope(), ctx_->exec_cfg()};
+        auto body = pop_scope();
+        if (slm_byte_offset() > 0) {
+            auto slm_buf = var_t::make(u8[slm_byte_offset()].with_slm(), "slm");
+            auto slm_alloc = builtin_t::make("alloc")(slm_buf);
```

Comment:
Would it make sense to modify `slm_byte_offset()` to be an `expr_t` an export it via `interface_` so that we can support dynamic slm sizing?